src/lib.rs

```diff
@@ -272,6 +272,31 @@ pub struct PaintCtx<'a, 'b: 'a> {
     /// The render context for actually painting.
     pub render_ctx: &'a mut Piet<'b>,
     pub window_id: WindowId,
+    /// The currently visible region.
+    pub(crate) region: Region,
+}
+
+/// A region of a widget, generally used to describe what needs to be drawn.
+#[derive(Debug, Clone)]
+pub struct Region(Rect);
+
+impl Region {
+    /// Returns the smallest `Rect` that encloses the entire region.
+    pub fn to_rect(&self) -> Rect {
+        self.0
+    }
+
+    /// Returns `true` if `self` intersects with `other`.
+    #[inline]
+    pub fn intersects(&self, other: Rect) -> bool {
```

Comment:
I was thinking `impl Into<Region>` but maybe keeping it simpler is better. It's the kind of thing I can imagine wanting specialization when that lands...

Reply:
My thinking here was that I would then have to implement intersection for every concrete type that impls `Into<Region>`; but I think we can revisit when we get there?